scripts/west_commands/boards.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import collections
+import os
+import re
+import shutil
+import subprocess
+import sys
+import textwrap
+
+from west import log
+from west import cmake
+from west.configuration import config
+from west.commands import WestCommand
+from cmake import run_cmake
+
+BOARDS_DESCRIPTION = '''\
+Display information about boards
+'''
+
+class Boards(WestCommand):
+
+    def __init__(self):
+        super(Boards, self).__init__(
+            'boards',
+            # Keep this in sync with the string in west-commands.yml.
+            'display information about supported boards',
+            BOARDS_DESCRIPTION,
+            accepts_unknown_args=False)
+
+    def do_add_parser(self, parser_adder):
+        default_fmt = '{name:32} ({arch})'
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description=self.description,
+            epilog=textwrap.dedent('''\
+            FORMAT STRINGS
+
+            Boards are listed using a Python 3 format string. Arguments
+            to the format string are accessed by name.
+
+            The default format string is:
+
+            "{}"
+
+            The following arguments are available:
+
+            - name: board name
+            - arch: board architecture
```

Comment:
It's a shame that the boards.cmake output is so limited. It would be nice to have `{abspath}` and `{posixpath}` format strings which print the path to the board directory just like we can do for projects in `west list`.

Subject for another PR, I guess. Could also make the output easier to parse if boards.cmake were more flexible in what it printed also.

Reply:
Let's leave this for another PR. I really don't want to overcomplicate things at this stage and `boards.cmake` needs a bit of rework down the line as you mention.

Reply:
Note that it might be simplify things to port boards.cmake to python.

Reply:
> Note that it might be simplify things to eventually port boards.cmake to python.

True.

Yep, definitely future work.